### PR TITLE
[DNM] Allowing players to cryo others. 

### PR DESCRIPTION
**Pull request by Ironnhawk**
## About The Pull Request

Gives players the ability to cryo other players as long as they have been SSD for more then 30 minutes. 

## How This Contributes To The Skyrat Roleplay Experience

Requires less admin intervention. 

## Changelog

:cl:
add: Added new mechanics or gameplay changes
add: Added more things
expansion: Expands content of an existing feature
del: Removed old things
qol: made something easier to use
balance: rebalanced something
fix: fixed a few things
soundadd: added a new sound thingy
sounddel: removed an old sound thingy
imageadd: added some icons and images
imagedel: deleted some icons and images
spellcheck: fixed a few typos
code: changed some code
refactor: refactored some code
config: changed some config setting
admin: messed with admin stuff
server: something server ops should know
/:cl:

### CODE REVIEW

**Comment by Ironnhawk** [comment]
So im not great at coding and have been using this as a learning experience. Currently, it does not work, I am trying to determine the issue but if anyone wants to give me a breakdown of what I did wrong, I would greatly appreciate it.

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 1]
```diff
@@ -340,12 +340,29 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 		return
 
 	if(target.key && user != target)
-		if(iscyborg(target))
+		var/mob/living/mob_occupant = target //Get mob living
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: the argument `target` is already typecasted to `mob/living`, uneeded new var

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 2]
```diff
@@ -340,12 +340,29 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
+		if (target.getorgan(/obj/item/organ/brain) ) //Target the Brain
+			if(ai_controller?.ai_status == AI_STATUS_ON) // Is the character empty / AI Controlled
+				for (mob_occupant) // Get Target Client
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: I don't quite get what you're doing here and I don't think it'll work

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 3]
```diff
@@ -340,12 +340,29 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
...
+					//if(round(((world.time - target.lastclienttime) / (1 MINUTES)),1) >= 1)
+					if(target.lastclienttime + 15 MINUTES <= world.time)
+						to_chat(user, span_danger("You can't put [target] into [src]. [target.p_theyre(capitalized = TRUE)] has not been asleep for 15 minutes."))
+						continue
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: why a continue and not a return?

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 4]
```diff
@@ -340,12 +340,29 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 		return
 
 	if(target.key && user != target)
-		if(iscyborg(target))
+		var/mob/living/mob_occupant = target //Get mob living
+		if (target.getorgan(/obj/item/organ/brain) ) //Target the Brain
+			if(ai_controller?.ai_status == AI_STATUS_ON) // Is the character empty / AI Controlled
+				for (mob_occupant) // Get Target Client
+					//if(round(((world.time - target.lastclienttime) / (1 MINUTES)),1) >= 1)
+					if(target.lastclienttime + 15 MINUTES <= world.time)
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm]: 15 MINUTES would be better in a macro, no magic numbers

File: modular_skyrat/modules/cryosleep/code/cryopod.dm [hunk 5]
```diff
@@ -340,12 +340,29 @@ GLOBAL_LIST_EMPTY(cryopod_computers)
 		return
 
 	if(target.key && user != target)
-		if(iscyborg(target))
+		var/mob/living/mob_occupant = target //Get mob living
+		if (target.getorgan(/obj/item/organ/brain) ) //Target the Brain
+			if(ai_controller?.ai_status == AI_STATUS_ON) // Is the character empty / AI Controlled
```
Comment (Zonespace27) [inline modular_skyrat/modules/cryosleep/code/cryopod.dm:345]: Check for a lack of client in an OR statement. This also allows for cryoing of `/mob/living`s that aren't animals but are AI controlled.

**Review by Zonespace27 — COMMENTED** [review]
Since you asked for a breakdown of why it wasn't working, here's a review.

File: modular_skyrat/modules/cryosleep/code/cryopod.dm (hunk 1 above)
Comment (Ironnhawk) [reply modular_skyrat/modules/cryosleep/code/cryopod.dm]: Removed that var <3

File: modular_skyrat/modules/cryosleep/code/cryopod.dm (hunk 2 above)
Comment (Ironnhawk) [reply modular_skyrat/modules/cryosleep/code/cryopod.dm]: So i based this off of the Examine code, It checks the target for a brain, then checks if it has AI_Stats_On suggesting it is no longer player controlled. I removed the For Occupant and Var as they arent needed.

**Comment by Gandalf2k15** [comment]
reeeeopen when you're done with the changes

**Pull request closed without merging**